specreduce/extract.py

```diff
@@ -123,6 +123,34 @@ def _to_spectrum1d_pixels(fluxes):
                       flux=fluxes)
 
 
+def _align_along_trace(img, array_trace, disp_axis=1, crossdisp_axis=0):
+    """
+    Given an arbitrary trace ``array_trace`` (an np.ndarray), roll
+    all columns of ``nddata`` to shift the NDData's pixels nearest
+    to the trace to the center of the spatial dimension of the
+    NDData.
+    """
+    # TODO: this workflow does not support extraction for >2D spectra
+    if not (disp_axis == 1 and crossdisp_axis == 0):
+        # take the transpose to ensure the rows are the cross-disp axis:
+        img = img.T
+
+    n_rows = img.shape[0]
```

Comment:
```suggestion
    n_rows = img.shape[crossdisp_axis]
```
You could also add `n_cols` so operations later on look more consistent.

Reply:
I've changed this expression in 078b48d.